gnuradio-runtime/lib/block_executor.cc

```diff
@@ -260,17 +289,40 @@ block_executor::state block_executor::run_one_iteration()
         d_start_nitems_read.resize(0);
 
         // determine the minimum available output space
-        noutput_items =
-            min_available_space(d, m->output_multiple(), m->min_noutput_items());
+        output_idx = 0;
+    out_try_again:
+        noutput_items = min_available_space(
+            m, d, m->output_multiple(), m->min_noutput_items(), output_idx);
         noutput_items = std::min(noutput_items, max_noutput_items);
         LOG(std::ostringstream msg; msg << "source:  noutput_items = " << noutput_items;
             GR_LOG_INFO(d_debug_logger, msg.str()););
         if (noutput_items == -1) // we're done
             goto were_done;
 
         if (noutput_items == 0) { // we're output blocked
-            LOG(GR_LOG_INFO(d_debug_logger, "BLKD_OUT"););
-            return BLKD_OUT;
+            LOG(std::ostringstream msg; msg << m << " -- BLKD_OUT";
+                GR_LOG_INFO(d_debug_logger, msg.str()););
+
+            buffer_sptr out_buf = d->output(output_idx);
+
+            if (out_buf->output_blkd_cb_ready(m->output_multiple())) {
+                gr::custom_lock lock(std::ref(*out_buf->mutex()), out_buf);
+                if (!out_buf->output_blocked_callback(m->output_multiple(),
+                                                      m->min_noutput_items())) {
```

Comment:
Why is the second argument a bool, but int (min_noutput_items()) is input to the output_blocked_callback function

Reply:
Whoops... the min number of outputs items was passed in as a parameter to the callback at one point but was later found to not be needed and removed.  Somehow I missed this during cleanup. An update will be forthcoming.